### ISSUE TITLE
setting zf?

### ISSUE THREAD

**Issue by zonkzonk** [radare/radare2#4371]
morrn,

as asked in #radare, how to get/set zf bit? I can get fine with dr zf or dr rflags, but how to set?

```
,r2  -qc "dr zf=0x00000200;??" -dw use
Process with PID 1514 started...
attach 1514 1514
bin.baddr 0x00600000
Assuming filepath ./use
Warning: Cannot initialize dynamic strings
asm.bits 64
1
```

Greetings
--zlul

![use](https://cloud.githubusercontent.com/assets/5694980/13909263/c82210ba-ef0f-11e5-92a2-90b93e26f682.png)

**Comment by radare** [comment]
dr zf=1

But this depends on the regprofile so you will probably need to fix it (see drp) or just modify the eflags

**Comment by zonkzonk** [comment]
hm, doesn't work:

[0x7f57de928d90]> drp~zf
gpr     zf      .1      .1158   0       zero

**Comment by zonkzonk** [comment]
using dr zf=1 at this time :)

**Comment by zonkzonk** [comment]
exit code ?? seems wrong in current version:
[0x00600078]> dr zf=1
[0x00600078]> ??
1
[0x00600078]> !r2 -v
Segmentation fault
,r2 -v
radare2 0.10.2-git 10652 @ linux-little-x86-64 git.0.10.1-195-g2a82395
commit: 2a82395c3fc0b23a1376acfa14cc308d47f297cf build: 2016-03-21

**Comment by radare** [comment]
Ive got a segfault doing !vim yesterday.. I think theres a regression in the code that setups the environ vars.

You can also show the version with the ?V command

**Comment by zonkzonk** [comment]
,file core.1164*
core.11648: ELF 64-bit LSB core file x86-64, version 1 (SYSV),
SVR4-style, from 'r2 -d use'
core.11649: ELF 64-bit LSB core file x86-64, version 1 (SYSV),
SVR4-style, from './use'
,gdb -q r2 core.11648
Reading symbols from r2...done.

warning: core file may not match specified executable file.
[New LWP 11648]

warning: Could not load shared library symbols for linux-vdso.so.1.
Do you need "set solib-search-path" or "set sysroot"?
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Core was generated by `r2 -d use'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f29e8a46646 in strlen () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007f29e8a46646 in strlen () from /usr/lib/libc.so.6
#1  0x00007f29e89fd809 in __add_to_environ () from /usr/lib/libc.so.6
#2  0x00007f29e93a924d in r_sys_setenv (key=0x7f29ed046258 "R2_COLOR",
value=0x1 <error: Cannot access memory at address 0x1>
#3  0x00007f29ecfd067c in r_core_sysenv_begin (core=0x55c0665738e0
<r>, cmd=0x55c06734d031 "r2 -v") at file.c:236
#4  0x00007f29ecfcaf0a in cmd_system (data=0x55c0665738e0 <r>,
input=0x55c06734d031 "r2 -v") at cmd.c:1049
#5  0x00007f29ecfff1c1 in r_cmd_call (cmd=0x55c0673a3a90,
input=0x55c06734d030 "!r2 -v") at cmd_api.c:210
#6  0x00007f29ecfcd520 in r_core_cmd_subst_i (core=0x55c0665738e0 <r>,
cmd=0x55c06734d030 "!r2 -v", colon=0x0) at cmd.c:1752
#7  0x00007f29ecfcb6b4 in r_core_cmd_subst (core=0x55c0665738e0 <r>,
cmd=0x55c06734d030 "!r2 -v") at cmd.c:1213
#8  0x00007f29ecfcea70 in r_core_cmd (core=0x55c0665738e0 <r>,
cstr=0x55c0673fe340 "!r2 -v", log=1) at cmd.c:2158
#9  0x00007f29ecf7eedd in r_core_prompt_exec (r=0x55c0665738e0 <r>) at
core.c:1505
#10 0x000055c066371623 in main (argc=3, argv=0x7ffc09e171e8,
envp=0x7ffc09e17208) at radare2.c:853
(gdb) i r
rax            0x1      1
rbx            0x8      8
rcx            0x1      1
rdx            0x1500   5376
rsi            0x1      1
rdi            0x1      1
rbp            0x7ffc09e16260   0x7ffc09e16260
rsp            0x7ffc09e161f8   0x7ffc09e161f8
r8             0xffffff00       4294967040
r9             0x55c0673fef30   94284854325040
r10            0x55c0673fef00   94284854324992
r11            0x4      4
r12            0x55c06636ed40   94284836957504
r13            0x7ffc09e171e0   140720474255840
r14            0x0      0
r15            0x0      0
rip            0x7f29e8a46646   0x7f29e8a46646 <strlen+38>
eflags         0x10293  [ CF AF SF IF RF ]
cs             0x33     51
ss             0x2b     43
ds             0x0      0
es             0x0      0
fs             0x0      0
gs             0x0      0

**Comment by zonkzonk** [comment]
can be closed, can't reproduce in bbbe3a2519b53e85781d1f9d3f082c9318eba5df
